src/template.rs

```diff
@@ -279,6 +280,7 @@ impl Template {
 
         let mut buffer: String = String::new();
         let mut state = ParserState::Text;
+        let mut old_state = ParserState::Text;
```

Comment:
To be honest I couldn't  figure out if we need to track the old state or
not.

I was trying to preserve as much of the parser state as possible for after.

What do you think?